dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1404,12 +1404,20 @@ public String getWorkflowDag(@ApiParam(hidden = true) @Auth Optional<User> user,
         WorkflowVersion workflowVersion = getWorkflowVersion(workflow, workflowVersionId);
         SourceFile mainDescriptor = getMainDescriptorFile(workflowVersion);
 
+        if (workflowVersion.getDagJson() != null) {
+            return workflowVersion.getDagJson();
+        }
+
         if (mainDescriptor != null) {
             Set<SourceFile> secondaryDescContent = extractDescriptorAndSecondaryFiles(workflowVersion);
 
             LanguageHandlerInterface lInterface = LanguageHandlerFactory.getInterface(workflow.getFileType());
-            return lInterface.getCleanDAG(workflowVersion.getWorkflowPath(), mainDescriptor.getContent(), secondaryDescContent,
+            String dagJson = lInterface.getCleanDAG(workflowVersion.getWorkflowPath(), mainDescriptor.getContent(), secondaryDescContent,
                     LanguageHandlerInterface.Type.DAG, toolDAO);
+            if (workflowVersion.getReferenceType() == Version.ReferenceType.TAG && workflowVersion.isValid()) {
```

Comment:
Why only do this if `isValid()` is true? We would want to avoid parsing invalid workflows again and again as well, I would think. Or am I missing something?

Reply:
I guess I was thinking a user would be more likely to recreate a tag if it wasn't valid on our end and to take up less space in the db. Not that it takes up much space (see description). But yeah, other than that, not really much of a reason so will change to not check validity.